microsoft/PowerToys
# Save QueryHistory, UserSelectedRecord after each update

## Summary of the Pull Request

**What is this about:**
Save QueryHistory and UserSelectedRecord after an update. It happens only when a user opens a search result in PT Run.
Delete TopMostRecord as it is not used anywhere. 

**What is include in the PR:** 
- Save QueryHistory and UserSelectedRecord after each update
- Escape info in `QueryHistory.json` and delete `UserSelectedRecord.json` in BugReportTool
- Delete `TopMostRecord`

**How does someone test / validate:** 
- Open something through PT Run
- Check that `PowerToys Run\Settings\QueryHistory.json` and `PowerToys Run\Settings\UserSelectedRecord.json` changed
- Run BugReportTool and check that `QueryHistory.json` escaped information and `UserSelectedRecord.json` is not present

## Quality Checklist

- [X] **Linked issue:** #9098

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
